aten/src/ATen/native/cuda/EmbeddingBackwardKernel.cu

```diff
@@ -0,0 +1,311 @@
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/NativeFunctions.h>
+
+#include <ATen/AccumulateType.h>
+
+#include <THC/THCDeviceUtils.cuh>
+#include <THC/THCTensorMathReduce.cuh>
+#include <THC/THCTensorSort.cuh>
+#include <THC/THCThrustAllocator.cuh>
+#include <THC/THCAtomics.cuh>
+
+#include <thrust/execution_policy.h>
+#include <thrust/unique.h>
+#include <thrust/device_vector.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+// The maximum block size in CUDA
+constexpr int MAX_BLOCK_SIZE = 1024;
+// Number of rows each thread should sum initially
+constexpr int NROWS_PER_THREAD = 10;
+
+#ifdef __HIP_PLATFORM_HCC__
+    constexpr int WARP_SIZE = 64;
+#else
+    constexpr int WARP_SIZE = 32;
+#endif
+
+// Fast ceil division (no overflow checking)
+__host__ __device__ __forceinline__
+int64_t ceil_div(int64_t x, int64_t y) {
+  return (x + y - 1) / y;
+}
+
+__global__
+void split_segment_sizes_kernel(int64_t *ret, const int64_t *segment_offsets,
+                                int64_t num_of_segments, int64_t blocksize, int64_t numel) {
```

Comment:
My first intuition of seeing `blocksize` is that it refers the CUDA block size, but it is actually `NROWS_PER_THREAD`. Can we come up with a different name to avoid conflicting with CUDA block?

Reply:
Agree, it is now `constexpr`